### PR TITLE
CO2 capture system, piperazine surrogate models (from reference)

### PR DESCRIPTION
**Pull request by MAZamarripa**
## Fixes
creating a surrogate model of a carbon capture system (solvent-based piperazine)

## Summary/Motivation:
creating a surrogate model of a carbon capture system (solvent-based piperazine)
surrogate models obtained from a reference (open source model)

## Changes proposed in this PR:
- adding co2 system surrogate (mass balances + surrogates for SRD)
- adding test 

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/power_generation/flowsheets/sofc/surrogates/cpu.py [hunk 1]
```diff
@@ -7,8 +7,20 @@
 # Washington to design NGFC systems with high efficiencies and low CO2
 # emissions.
 ##############################################################################
+##############################################################################
```
Comment (jghouse88) [inline idaes/power_generation/flowsheets/sofc/surrogates/cpu.py:10]: Is this file intended in this PR?

File: idaes/power_generation/flowsheets/sofc/surrogates/cpu_surrogate_methods.py [hunk 2]
```diff
@@ -1,3 +1,16 @@
+##############################################################################
```
Comment (jghouse88) [inline idaes/power_generation/flowsheets/sofc/surrogates/cpu_surrogate_methods.py:1]: Same comment: is this file intended to be part of this PR?

File: idaes/power_generation/flowsheets/sofc/surrogates/cpu.py (hunk 1 above)
Comment (MAZamarripa) [reply idaes/power_generation/flowsheets/sofc/surrogates/cpu.py:10]: I had to fix those files to pass the tests

File: idaes/power_generation/flowsheets/sofc/surrogates/cpu_surrogate_methods.py (hunk 2 above)
Comment (MAZamarripa) [reply idaes/power_generation/flowsheets/sofc/surrogates/cpu_surrogate_methods.py:1]: I had to fix those files to pass the tests

File: idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+                         self.component_list,
+                         doc="Inlet component mole flow eqn")
+        def flow_mol_comp_inlet_eqn(b, t, c):
+            if c == "H2O":
+                return b.inlet_flow_mol_comp[t, c] == \
+                    b.exhaust_gas_flow_mol_comp[t, c] + b.water_drop[t]
+            elif c == "CO2":
+                return b.inlet_flow_mol_comp[t, c] == \
+                    b.exhaust_gas_flow_mol_comp[t, c] \
+                    + b.pureCO2_flow_mol_comp[t, "CO2"]
```
Comment (nareshsusarla) [inline idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py:215]: "CO2" can be the index c for consistancy

File: idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+    def add_material_balances(self):
+        ''' This section is for material balance constraints'''
+
+        # pureCO2 mass balance
+        @self.Constraint(self.flowsheet().config.time,
+                         self.component_list,
+                         doc="pureCO2 mass balances")
+        def pureCO2_eqn(b, t, c):
+            if c == "CO2":
+                return b.pureCO2_flow_mol_comp[t, "CO2"] == \
```
Comment (nareshsusarla) [inline idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py:194]: 'if' condition looks redundant

File: idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+
+        Returns:
+            None
+        '''
+        iscale.calculate_scaling_factors(blk)  # remove to solve using baron
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(blk.name, outlvl, tag="unit")
+        opt = pyo.SolverFactory(solver)
```
Comment (nareshsusarla) [inline idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py:299]: will it be better to use the '`get_solver`' method here? that will also change the args in `initialize `method `solver=None, optarg=None`

**Review by nareshsusarla — APPROVED** [review]
looks good, some minor comments to consider

File: idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+                                    doc='PureCO2 pressure [Pa]')
+
+        self.exhaust_gas_pressure = Var(self.flowsheet().config.time,
+                                        initialize=17,
+                                        units=pressure_units,
+                                        doc='exhaust_gas pressure [Pa]')
+        # CO2 Capture rate
+        self.CO2_capture_rate = Var(self.flowsheet().config.time,
+                                    initialize=0.9,
+                                    doc='CO2 capture rate')
```
Comment (nareshsusarla) [inline idaes/power_generation/carbon_capture/piperazine_surrogates/co2_capture_system.py:167]: If the surrogate is only valid for CO2_capture_rate = 0.9, can we add a note here?

**Comment by andrewlee94** [comment]
Closing this for now due to lack of activity and the existing conflicts.
